corehq/apps/tour/models.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+from django.contrib.auth.models import User
+from django.db import models
+import json_field
+
+
+class GuidedTours(models.Model):
+    user = models.ForeignKey(
+        User,
+        on_delete=models.PROTECT,
+        unique=True
+    )
+    seen_tours = json_field.JSONField(
+        default={},
+    )
+
+    @classmethod
+    def has_seen_tour(cls, user, tour_slug):
+        guided_tour, _ = GuidedTours.objects.get_or_create(user=user)
+        return tour_slug not in guided_tour.seen_tours
+
+    @classmethod
+    def mark_as_seen(cls, user, tour_slug):
+        guided_tour, _ = GuidedTours.objects.get_or_create(user=user)
+        guided_tour.seen_tours[tour_slug] = datetime.datetime.now()
+        guided_tour.save()
+
+
+def mark_tour_as_seen_for_user(user, tour_slug):
+    GuidedTours.mark_as_seen(user, tour_slug)
```

Comment:
previous feedback was to abstract these methods in case we decide to change how we handle saving the tour.